src/js/tracking.js

```diff
@@ -15,8 +15,16 @@ class Tracker {
         }
     }
     initialize(trackingCode) {
-        ReactGA.initialize(trackingCode);
-        this.initialized = true;
+        let self = this;
+        return new Promise(function(resolve, reject) {
+            if (self.initialized) {
+                reject(self);
+            } else {
+                ReactGA.initialize(trackingCode);
+                self.initialized = true;
+                resolve(self);
+            }
+        });
```

Comment:
since there is nothing async happening, wouldn't:
```suggestion
        if (self.initialized) {
          return;
        }
        ReactGA.initialize(trackingCode);
        self.initialized = true;
```
be enough? or does the initialization take some time to complete?

Reply:
@mzedel You are right, no need to go async. We still need to run the "set" functions conditionally, because we only need them once. I updated the code to return a boolean if the initialization is performed.